docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc

```diff
@@ -0,0 +1,358 @@
+
+* `ssh`
+* `curl`
+* `kubectl`
+* `yq` https://mikefarah.gitbook.io/yq[yq YAML processor] (version 4 or higher)
+* `whois`
+
+== Prepare IP range
+
+. Get a suitable public IP range.
+You can find all IP ranges owned by VSHN
++
+[source,console]
+----
+whois -i mnt-by MNT-VSHN -T inetnum -r
+----
+Please coordinate with other VSHNeers before using an IP range.
+
+. Update RIPE DB to reflect the new usage of the IP range.
+
+. Create a ticket at cloudscale.ch to make the range available as a Floating Network in the project of the cluster.
+
+== Configure Loadbalancers
+
+Add the chosen IP range as a loadbalancer address pool in the https://git.vshn.net/appuio/appuio_hieradata/-/tree/master/lbaas[APPUiO hieradata].
+Assuming you want to add the IP range `203.0.113.128/25` to the cluster `c-cold-bush-7254`:
+
+[source,console]
+----
+export CLUSTER_ID="c-cold-bush-7254"
+export IP_RANGE="203.0.113.128/25"
+
+mkdir -p lbaas/${CLUSTER_ID}
+touch lbaas/${CLUSTER_ID}/lb.yaml
+yq eval -i '."profile_openshift4_gateway::loadbalancer_ip_address_pools" += [${IP_RANGE}]' lbaas/${CLUSTER_ID}/lb.yaml
+
+git commit -a -m "Add loadbalancer IP range ${IP_RANGE} to ${CLUSTER_ID}"
+git push
+----
+ 
+
+=== Setup BGP speaker
+
+The following steps need to be done manually on both loadbalancer VMs
+
+. Install FRR
++
+[source,console]
+----
+apt install frr
+----
+
+. Enable bgp by setting `bgpd=yes` in `/etc/frr/daemon`
++
+[source]
+----
+...
+bgpd=yes
+ospfd=no
+ospf6d=no
+ripd=no
+ripngd=no
+...
+----
+
+. Restart FRR
++
+[source,console]
+----
+systemctl restart frr
+----
+  
+. Configure FRR
++
+Enter the FRR shell with
++
+[source,console]
+----
+vtysh
+----
++
+Configure BGP neighbors
++
+[source]
+----
+conf t
+router bgp 64512
+  neighbor <ip-infra-node-0> remote-as 64512 <1>
+  neighbor <ip-infra-node-1> remote-as 64512
+  neighbor <ip-infra-node-2> remote-as 64512
+  end
+write <2>
+----
+<1> You can add any of the cluster node IPs.
+We propose to peer with each of the infrastructure nodes.
+<2> You need to write the configuration to persist 
+
+== Configure Cilium
+
+. Add a ConfigMap called `bgp-config` in the namespace `cilium`.
+This ConfigMap contains https://metallb.universe.tf/usage/#bgp[MetalLB] configuration.
++
+Here's an example ConfigMap.
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: bgp-config
+  namespace: kube-system
+data:
+  config.yaml: |
+    peers:
+      - peer-address: 10.0.0.1 <1>
+        peer-asn: 64512
+        my-asn: 64512
+      - peer-address: 10.0.0.2 <1>
+        peer-asn: 64512
+        my-asn: 64512
+    address-pools:
+      - name: default
+        protocol: bgp
+        addresses:
+          - 203.0.113.128/25 <2>
+----
+<1> IP of the loadbalancers
+<2> Public IP range chosen
+
+. Update the Cilium configuration to enable BGP.
+Our Cilium setup is configured through the `CiliumConfig` CRD called `cilium` in namespace `cilium`.
+The CRD contains helm values.
+Update the CRD `spec` with
```

Comment:
Might be worth it to craft a `kubectl patch` which can be copy-pasted for this step

Reply:
Again I don't know if it makes sense to improve this, as this should be managed by the Cilium component very soon.